fluss-server/src/main/java/org/apache/fluss/server/metadata/TabletServerMetadataCache.java

```diff
@@ -122,18 +120,12 @@ public TableMetadata getTableMetadata(TablePath tablePath) {
         OptionalLong tableIdOpt = snapshot.getTableId(tablePath);
         List<BucketMetadata> bucketMetadataList;
         if (!tableIdOpt.isPresent()) {
-            // TODO no need to get assignment from zk if refactor client metadata cache. Trace by
-            // https://github.com/apache/fluss/issues/483
-            // get table assignment from zk.
-            bucketMetadataList =
-                    getTableMetadataFromZk(
-                            zkClient, tablePath, tableInfo.getTableId(), tableInfo.isPartitioned());
-        } else {
-            // get table assignment from cache.
-            bucketMetadataList =
-                    new ArrayList<>(
-                            snapshot.getBucketMetadataForTable(tableIdOpt.getAsLong()).values());
+
+            return null;
```

Comment:
Add `@Nullable` annotation to the return type of the method, or return `Optional` directly (it seems it is used as `Optional` later).